apps/agreement/contracts/Agreement.sol

```diff
@@ -0,0 +1,1338 @@
+        DisputableInfo storage disputableInfo = disputableInfos[_disputable];
+        activated = disputableInfo.activated;
+        uint256 length = disputableInfo.nextCollateralRequirementsId;
+        currentCollateralRequirementId = length == 0 ? 0 : length - 1;
+    }
+
+    /**
+    * @dev Tell the information related to a collateral requirement of a disputable app
+    * @param _disputable Address of the disputable app querying the collateral requirements of
+    * @param _collateralRequirementId Identification number of the collateral being queried
+    * @return collateralToken Address of the ERC20 token to be used for collateral
+    * @return actionAmount Amount of collateral tokens that will be locked every time an action is created
+    * @return challengeAmount Amount of collateral tokens that will be locked every time an action is challenged
+    * @return challengeDuration Challenge duration in seconds, during which the submitter can raise a dispute
+    */
+    function getCollateralRequirement(address _disputable, uint256 _collateralRequirementId) external view
+        returns (
+            ERC20 collateralToken,
+            uint256 actionAmount,
+            uint256 challengeAmount,
+            uint64 challengeDuration
+        )
+    {
+        DisputableInfo storage disputableInfo = disputableInfos[_disputable];
+        CollateralRequirement storage collateral = disputableInfo.collateralRequirements[_collateralRequirementId];
+        collateralToken = collateral.token;
+        actionAmount = collateral.actionAmount;
+        challengeAmount = collateral.challengeAmount;
+        challengeDuration = collateral.challengeDuration;
+    }
+
+    /**
+    * @dev Tell the information related to an action
+    * @param _actionId Identification number of the action being queried
+    * @return disputable Address of the disputable that created the action
+    * @return disputableActionId Identification number of the action in the context of the disputable
+    * @return collateralRequirementId Identification number of the collateral requirements applicable to the action
+    * @return settingId Identification number of the agreement setting at the moment the action was submitted
+    * @return submitter Address that has submitted the action
+    * @return closed Whether the action was manually closed by the disputable app
+    * @return context Link to a human-readable text providing context for the action
+    * @return currentChallengeId Identification number of the current challenge for the action
+    */
+    function getAction(uint256 _actionId) external view
+        returns (
+            address disputable,
+            uint256 disputableActionId,
+            uint256 collateralRequirementId,
+            uint256 settingId,
+            address submitter,
+            bool closed,
+            bytes context,
+            uint256 currentChallengeId
+        )
+    {
+        Action storage action = actions[_actionId];
```

Comment:
What I meant in a previous comment about erroring if something doesn't exist was meant for the getters: https://github.com/aragon/aragon-apps/pull/1133/files/f698674b346af5052bcc488df8eb51ddbb8f3f5f#diff-9ebddb6de87cd26830aa77a8db008ced.

In the past we have added checks in getters to revert if the id doesn't exist yet. However, I'm starting to question that decision since it might lead to a large bytecode increase (at least with modifiers).